### PR TITLE
Send update events from service-controller.

### PR DESCRIPTION
**Pull request by bprashanth**
@maisem since I noticed you took https://github.com/kubernetes/kubernetes/issues/20547

### CODE REVIEW

File: pkg/controller/service/servicecontroller.go [hunk 1]
```diff
@@ -457,24 +457,31 @@ func (s *serviceCache) delete(serviceName string) {
...
 	if wantsLoadBalancer(oldService) != wantsLoadBalancer(newService) {
+		s.eventRecorder.Eventf(newService, api.EventTypeNormal, "Type", "%v -> %v",
+			oldService.Spec.Type, newService.Spec.Type)
 		return true
 	}
 	if !portsEqualForLB(oldService, newService) || oldService.Spec.SessionAffinity != newService.Spec.SessionAffinity {
 		return true
 	}
 	if !loadBalancerIPsAreEqual(oldService, newService) {
+		s.eventRecorder.Eventf(newService, api.EventTypeNormal, "Loadbalancer IP", "%v -> %v",
```
Comment (mikedanese) [inline pkg/controller/service/servicecontroller.go]: Reason should be camel case https://github.com/kubernetes/kubernetes/blob/master/docs/devel/api-conventions.md#events

**Comment by mikedanese** [comment]
These could all have the same reason UpdatingService and describe why in the message

File: pkg/controller/service/servicecontroller.go [hunk 2]
```diff
@@ -457,24 +457,31 @@ func (s *serviceCache) delete(serviceName string) {
...
 		return true
 	}
 	if len(oldService.Spec.ExternalIPs) != len(newService.Spec.ExternalIPs) {
+		s.eventRecorder.Eventf(newService, api.EventTypeNormal, "External IP count", "%v -> %v",
+			len(oldService.Spec.ExternalIPs), len(newService.Spec.ExternalIPs))
 		return true
 	}
 	for i := range oldService.Spec.ExternalIPs {
 		if oldService.Spec.ExternalIPs[i] != newService.Spec.ExternalIPs[i] {
+			s.eventRecorder.Eventf(newService, api.EventTypeNormal, "Add external IP", newService.Spec.ExternalIPs[i])
```
Comment (mikedanese) [inline pkg/controller/service/servicecontroller.go]: This could use a better message

**Comment by bprashanth** [comment]
> These could all have the same reason UpdatingService and describe why in the message

I tried that initially and though having UpdateService and a message like foo changed from x -> y was redundant. Thoughts?

**Comment by mikedanese** [comment]
It would make searching/grepping somewhat easier to have a string that is common across all of these events. It's not super important

**Pull request merged**